frontend/src/components/accountPage/SSHKeysForm/SSHKeysForm.tsx

```diff
@@ -0,0 +1,124 @@
+import { Input, Form, Button, Row } from 'antd';
+import { FC, useState } from 'react';
+
+export interface ISSHKeysFormProps {
+  onSaveKey: (newKey: { name: string; key: string }) => void;
+  onCancel: () => void;
+}
+
+const acceptedAlgorithms = [
+  'sk-ecdsa-sha2-nistp256@openssh.com',
+  'ecdsa-sha2-nistp256',
+  'ecdsa-sha2-nistp384',
+  'ecdsa-sha2-nistp521',
+  'sk-ssh-ed25519@openssh.com',
+  'ssh-ed25519',
+  'ssh-dss',
+  'ssh-rsa',
+];
+
+const SSHKeysForm: FC<ISSHKeysFormProps> = props => {
+  const { onSaveKey, onCancel } = props;
+  const [validForm, setValidForm] = useState<boolean>(false);
+  const [form] = Form.useForm();
+
+  const cancelForm = () => {
+    form.resetFields();
+    onCancel();
+  };
+  const validateSSHKey = async (rules: any, key: string) => {
+    if (!key) throw new Error('Key field is mandatory');
+    const result = key.split(/\s+/); // split regardless the amount of spaces
+    if (result.length !== 2 && result.length !== 3)
+      throw new Error('Invalid key format');
+    if (result[1] === '') throw new Error('Empty SSH key');
+    if (!acceptedAlgorithms.includes(result[0]))
+      throw new Error('Unrecognized SSH key format');
+  };
+
+  // Used to disable Save button
+  const handleChange = async (
+    _: any,
+    { name, key }: { name: string; key: string }
+  ) => {
+    try {
+      if (key !== undefined && name !== undefined) {
```

Comment:
```suggestion
      if (key && name) {
```
Should be enough, better to test it though